lib/net/http.rb

```diff
@@ -1033,10 +1033,14 @@ def connect
           end
         end
         @ssl_context.set_params(ssl_parameters)
-        @ssl_context.session_cache_mode =
-          OpenSSL::SSL::SSLContext::SESSION_CACHE_CLIENT |
-          OpenSSL::SSL::SSLContext::SESSION_CACHE_NO_INTERNAL_STORE
-        @ssl_context.session_new_cb = proc {|sock, sess| @ssl_session = sess }
+        unless @ssl_context.session_cache_mode.nil? # a dummy method on JRuby
```

Comment:
If JRuby doesn't support `@ssl_context.session_cache_mode`, maybe don't define the method and use `if defined?(@ssl_context.session_cache_mode)`.

Reply:
Yes, we probably should do that, but we implemented it as a no-op plus warning so we might be reluctant to remove the method all together.

Reply:
:+1: the reasoning being `session_cache_mode` does not affect the user-code much overall and thus it existed for some time and removing it might be considered breaking ... thus the `nil` compromise will probably stay with us.
smt like `session_new_cb`, while being "low" level, tends to get much more critical in user-code - if we had it as a dummy attr and the callback never fired it would be counter productive ...